app/controllers/editions_controller.rb

```diff
@@ -34,8 +39,32 @@ def linking
     render action: "show"
   end
 
-  def unpublish
-    render action: "show"
+  def confirm_unpublish
+    if redirect_url.blank? || validate_redirect(redirect_url)
+      render "secondary_nav_tabs/confirm_unpublish"
+    else
+      error_message = "Redirect path is invalid. #{description(resource)} can not be unpublished."
+      @resource.errors.add(:redirect_url, error_message)
+      render "show"
+    end
+  end
+
+  def process_unpublish
+    artefact = @resource.artefact
+
+    success = params["redirect_url"].strip.empty? ? UnpublishService.call(artefact, current_user) : UnpublishService.call(artefact, current_user, redirect_url)
+    if success
+      notice = "Content unpublished"
+      notice << " and redirected" if redirect_url.present?
+      flash[:success] = notice
+      redirect_to root_path
+    else
+      @resource.errors.add(:unpublish, downstream_error_message)
+      render "secondary_nav_tabs/confirm_unpublish"
+    end
+  rescue StandardError
+    @resource.errors.add(:unpublish, downstream_error_message)
+    render "secondary_nav_tabs/confirm_unpublish"
```

Comment:
We could avoid duplication of those two lines 
```
@resource.errors.add(:unpublish, downstream_error_message)
 render "secondary_nav_tabs/confirm_unpublish"
```
by extracting them into separate method